partitioner-config-service-impl/src/main/java/org/hypertrace/partitioner/config/service/store/PartitionerProfilesDocumentStore.java

```diff
@@ -0,0 +1,65 @@
+package org.hypertrace.partitioner.config.service.store;
+
+import com.google.common.collect.ImmutableList;
+import io.grpc.Status;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.hypertrace.core.documentstore.*;
+import org.hypertrace.core.documentstore.Collection;
+import org.hypertrace.partitioner.config.service.v1.PartitionerProfile;
+
+public class PartitionerProfilesDocumentStore implements PartitionerProfilesStore {
+  public static final String PARTITIONER_PROFILES = "partitioner_profiles";
+  private static final String PARTITIONER_PROFILE_NAME_FIELD = "name";
+
+  private final Collection collection;
+
+  public PartitionerProfilesDocumentStore(Datastore datastore) {
+    this.collection = datastore.getCollection(PARTITIONER_PROFILES);
+  }
+
+  @Override
+  public Optional<PartitionerProfile> getPartitionerProfile(String profile) throws Exception {
+    Query query = new Query();
+    query.setFilter(new Filter(Filter.Op.EQ, PARTITIONER_PROFILE_NAME_FIELD, profile));
+    try (CloseableIterator<Document> it = collection.search(query)) {
+      List<Document> list = ImmutableList.copyOf(it);
+      if (list.size() == 1) {
+        return Optional.of(PartitionerProfileDocument.fromJson(list.get(0).toJson()));
+      } else if (list.size() > 1) {
+        throw Status.INTERNAL
+            .withDescription("More than 1 PartitionerProfile documents returned")
+            .asRuntimeException();
+      }
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public List<PartitionerProfile> getAllPartitionProfiles() throws Exception {
+    try (CloseableIterator<Document> it = collection.search(new Query())) {
+      List<Document> list = ImmutableList.copyOf(it);
+      return list.stream()
+          .map(document -> PartitionerProfileDocument.fromJson(document.toJson()))
+          .collect(Collectors.toList());
+    }
+  }
+
+  @Override
+  public void putPartitionerProfiles(List<PartitionerProfile> partitionerProfiles)
+      throws Exception {
+    Map<Key, Document> map = new LinkedHashMap<>();
+    partitionerProfiles.forEach(
```

Comment:
up to you on preference, but since you're using streams for the other transformations, equivalent here would be something like below. I didn't use an ordered map because I didn't see any need for it, but you could switch up the collector to a different ordered one too (like `ImmutableMap.toImmutableMap`)
```java
partitionerProfiles.stream()
  .collect(Collectors.toUnmodifiableMap(PartitionerProfile:: getName, PartitionerProfileDocument::new));
```

Reply:
done